src/Xablu.WebApiClient/Services/GraphQL/MutationRequest.cs

```diff
@@ -0,0 +1,175 @@
+                    if (hasProperties)
+                    {
+                        LoadProperties(propType);
+                    }
+                }
+            }
+
+            if (propList.Any())
+            {
+                _propertyListList.Add(propList);
+            }
+        }
+
+        protected virtual void CurateProperties()
+        {
+            var parentTypeToExclude = typeof(List<>);
+            foreach (List<PropertyDetail> propertyList in _propertyListList.Where(list => list.Any()))
+            {
+                propertyList.RemoveAll(p => p.ParentName == parentTypeToExclude.Name);
+            }
+        }
+
+        private string QueryBuilder(Type property)
+        {
+            var variableInputName = (Attribute.GetCustomAttribute(property, typeof(VariableInputAttribute)) as VariableInputAttribute)?.ModelInputName;
+            if (string.IsNullOrEmpty(variableInputName))
+            {
+                var errorMessage = "No VariableInputAttribute found. Please ensure the model has been marked or the value is not null";
+                throw new RequestException(errorMessage);
+
+            }
+
+            var variableString = CreateVariableString();
+            var methodString = CreateInputString(Mutation, variableInputName);
+            var queryString = $"mutation{methodString}{variableString}";
+            return queryString;
+        }
+
+        private string CreateVariableString()
+        {
+            string queryString = "";
+
+            foreach (List<PropertyDetail> propertyList in _propertyListList.Where(list => list.Any()))
+            {
+                propertyList.Reverse();
+                foreach (PropertyDetail property in propertyList)
+                {
+                    queryString = queryString.Any() ? queryString.Insert(0, ToLowerFirstChar(property?.FieldName) + " ") : queryString.Insert(0, ToLowerFirstChar(property?.FieldName));
+                }
+                queryString = "{" + $"{queryString}" + "}";
+            }
+            return queryString + "}";
+        }
+
+        private string CreateInputString(MutationDetail mutationDetail, string variableInputName)
+        {
+            string inputString = "";
+            var parameterInputList = new List<string>();
+
+            if (Variables != null)
+            {
+                foreach (var variable in Variables.GetType().GetProperties())
+                {
+                    parameterInputList.Add(variable.Name);
+                }
+            }
+
+            if (mutationDetail != null && !string.IsNullOrEmpty(variableInputName))
+            {
+                var parameterInputName = parameterInputList.Count > 0 ? parameterInputList[0] : "mutationParameterInputName";
+                inputString = $"(${ToLowerFirstChar(parameterInputName)}: {variableInputName}!)" + $"{{{mutationDetail.MutationName}({mutationDetail.MutationParameterName}: ${parameterInputName})";
```

Comment:
Using an attribute specified by the developer for a different purpose is really confusing... can we use a generated name here? 

Also in the case of BooksQL I'm testing the following mutation:

![imagen](https://user-images.githubusercontent.com/12127846/72897446-c0133500-3d00-11ea-8e56-a0ba15fb35ae.png)

And your implementation forces me to wrap the variable in an object called `bookISBN`, otherwise it doesn't work:

![imagen](https://user-images.githubusercontent.com/12127846/72897313-7296c800-3d00-11ea-8ac4-14e5874f1653.png)

Can you please look at this so that this isn't necessary? Please also make sure you are able to run the BooksQL example locally.

Reply:
will work on this

Reply:
<img width="714" alt="Screenshot 2020-01-23 at 14 58 06" src="https://user-images.githubusercontent.com/22239299/72990659-d7c9e680-3df0-11ea-9825-f15cdcc3cddd.png">

As you can see you don't have to wrap the variables at all you can just do new { }

Reply:
could you elaborate on this:

Using an attribute specified by the developer for a different purpose is really confusing... can we use a generated name here?


I fixed the wrapping issue and also implemented a fix for the variable naming for the string.

Reply:
And yes we can generate any name by changing the assigned name for the variable object parameter in the constructor.